lib/common/common/src/cpu.rs

```diff
@@ -0,0 +1,41 @@
+#[cfg(target_os = "linux")]
+use thiserror::Error;
+#[cfg(target_os = "linux")]
+use thread_priority::{set_current_thread_priority, ThreadPriority, ThreadPriorityValue};
+
+#[derive(Error, Debug)]
+#[cfg(target_os = "linux")]
+pub enum ThreadPriorityError {
+    #[error("Failed to set thread priority: {0:?}")]
+    SetThreadPriority(thread_priority::Error),
+    #[error("Failed to parse thread priority value: {0}")]
+    ParseNice(&'static str),
+}
+
+/// On Linux, make current thread lower priority (nice: 10).
+#[cfg(target_os = "linux")]
+pub fn linux_low_thread_priority() -> Result<(), ThreadPriorityError> {
+    // 25% corresponds to a nice value of 10
+    set_linux_thread_priority(25)
+}
+
+/// On Linux, make current thread high priority (nice: -10).
+#[cfg(target_os = "linux")]
+pub fn linux_high_thread_priority() -> Result<(), ThreadPriorityError> {
+    // 75% corresponds to a nice value of 10
+    set_linux_thread_priority(75)
+}
```

Comment:
IIRC, negative values require root.

Reply:
https://manned.org/renice.1#head6

> Users other than the superuser may only alter the priority of processes they own. Furthermore, an unprivileged user can only increase the "nice value" (i.e., choose a lower priority) and such changes are irreversible unless (since Linux 2.6.12) the user has a suitable "nice" resource limit (see ulimit(1p) and getrlimit(2)).
>
> The superuser may alter the priority of any process and set the priority to any value in the range -20 to 19. Useful priorities are: 19 (the affected processes will run only when nothing else in the system wants to), 0 (the "base" scheduling priority), anything negative (to make things go very fast).

Reply:
Correct. I mentioned this elsewhere and do a soft-fail. But I guess it would be nice to mention it here as well.

Reply:
Didn't read the PR description, went straight to the code.... 🙈

Reply:
I've added an extra warning in bb8387e586c8429238f6d86575bed875652259f1. I assume that is enough.